src/Microsoft.Health.Dicom.Functions/Indexing/ReindexDurableFunction.Trigger.cs

```diff
@@ -0,0 +1,78 @@
+using Microsoft.Health.Dicom.Core.Features.Retrieve;
+using Microsoft.Health.Dicom.Functions.Indexing.Configuration;
+
+namespace Microsoft.Health.Dicom.Functions.Indexing
+{
+    /// <summary>
+    /// Represents the Azure Durable Functions that perform the re-indexing of previously added DICOM instances
+    /// based on new tags configured by the user.
+    /// </summary>
+    public partial class ReindexDurableFunction
+    {
+        private readonly ReindexConfiguration _reindexConfig;
+        private readonly IReindexStore _reindexStore;
+        private readonly IInstanceReindexer _instanceReindexer;
+        private readonly IAddExtendedQueryTagService _addExtendedQueryTagService;
+        private readonly IInstanceStore _instanceStore;
+        private readonly IExtendedQueryTagStore _extendedQueryTagStore;
+
+        public ReindexDurableFunction(IOptions<IndexingConfiguration> configOptions,
+            IAddExtendedQueryTagService addExtendedQueryTagService,
+            IReindexStore reindexStore,
+            IInstanceStore instanceStore,
+            IInstanceReindexer instanceReindexer,
+            IExtendedQueryTagStore extendedQueryTagStore)
+        {
+            EnsureArg.IsNotNull(configOptions, nameof(configOptions));
+            EnsureArg.IsNotNull(reindexStore, nameof(reindexStore));
+            EnsureArg.IsNotNull(instanceReindexer, nameof(instanceReindexer));
+            EnsureArg.IsNotNull(addExtendedQueryTagService, nameof(addExtendedQueryTagService));
+            EnsureArg.IsNotNull(instanceStore, nameof(instanceStore));
+            EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
+            _reindexConfig = configOptions.Value.Add;
+            _reindexStore = reindexStore;
+            _instanceReindexer = instanceReindexer;
+            _addExtendedQueryTagService = addExtendedQueryTagService;
+            _instanceStore = instanceStore;
+            _extendedQueryTagStore = extendedQueryTagStore;
+        }
+
+        /// <summary>
+        /// The http trigger to add extended Query tags
+        /// </summary>
+        /// <param name="request">The http request.</param>
+        /// <param name="client">The client.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>The task.</returns>
+        [FunctionName(nameof(StartAddingExtendedQueryTagHttpTriggerAsync))]
+        public async Task StartAddingExtendedQueryTagHttpTriggerAsync(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "Post", Route = "extendedquerytags")] HttpRequestMessage request,
+            [DurableClient] IDurableOrchestrationClient client,
+            ILogger logger)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+            EnsureArg.IsNotNull(client, nameof(client));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+            var extendedQueryTags = await request.Content.ReadAsAsync<IEnumerable<AddExtendedQueryTagEntry>>();
```

Comment:
* As my understand, reason for `List` is it provides more capability while deserialization need same effort as `IEnumerable`? 
*  Do you mean we ultimately reserialize JSON by ourselve instead of utilize `ReadAsAsync`?